amoro-format-hudi/src/main/java/org/apache/amoro/formats/hudi/HudiHadoopCatalog.java

```diff
@@ -155,6 +156,9 @@ public List<String> listTables(String database) {
         () -> {
           FileSystem fs = fs();
           Path databasePath = new Path(warehouse, database);
+          if (!fs.exists(databasePath)) {
```

Comment:
As the Javadoc of `FormatCatalog#listTables` says that it will throw `NoSuchDatabaseException` when the database does not exist, I think adding this here is fine.

Can we use the following `fs.listStatus()` to check if the file exists?

Reply:
fs.listStatus(Path f) throws FileNotFoundException when the path does not exist，